### ISSUE TITLE
FR: support user API keys

### ISSUE THREAD

**Issue by mmahut** [nrdxp/cfdyndns#9]
It would be nice to to use users API keys and not the global key.

**Comment by jhillyerd** [comment]
It looks like support is there in HEAD, but it has not been released?

**Comment by jhillyerd** [comment]
@colemickens I was able to build dfd6200bedc27ade55b24f396a773201c625de92 as a NixOS overlay without any updates/patches, so you should be OK to tag a new release even if you don't have time to update deps.

**Comment by sysr-q** [comment]
Just to add to what @jhillyerd said, if anyone wants a NixOS overlay for this, I've forked the repo and merged in #7 and #13 from here since @colemickens seems to be MIA. You can use the following overlay in your nix config (but haven't tested with API tokens, just the annoying global zone key still):

```nix
{ config, pkgs, lib, ... }:
{
nixpkgs.overlays = [ (
  self: super:
  {
    cfdyndns = super.cfdyndns.overrideAttrs (old: rec {
      src = super.fetchFromGitHub {
        owner = "sysr-q";
        repo = "cfdyndns";
        rev = "4e703506df0298423a79be3e0efec7ecf6ae8680";
        sha256 = "0plijgr5y58ir9mjvxgm6jszz90pd1g0qjf21z0v5xrzg6bs2sy1";
      };

      cargoDeps = old.cargoDeps.overrideAttrs (lib.const {
        name = "cfdyndns-0.0.3-vendor.tar.gz";
        inherit src;
        outputHash = "OLxvob50FXumoxVfj97lOwSDaXncWGZXTf0wltnReQo=";
      });
    });
  }
) ];

services.cfdyndns = {
  enable = true;
  # ... etc
};
}
```

**Comment by colemickens** [comment]
Hey @sysr-q,

I'm confused, I was able to just `cargo update` and I am infact using a regular User API token successfully (See my latest commit+tag).

If you are interested in maintaining this, or maybe you and @nrdxp, I'd be happy to redirect folks to your fork and we can update the package in nixpkgs to point at your fork....

**Comment by nrdxp** [comment]
Interesting that you pinged me, I just created a [branch](https://github.com/nrdxp/cfdyndns/tree/refactor) where I started some cleanups and updates :sweat_smile: 

once I implement a few more of the features I'd like, I'd be happy to take over maintainership, but it'll probably be a bit. The build failures on latest Rust and runtime issues are fixed already though.

On a somewhat orthogonal point, however, the NixOS module should probably also support API tokens as they are simpler and don't require exposing an email as well.
